components/api/src/main/java/com/hotels/styx/api/FullHttpRequest.java

```diff
@@ -55,7 +55,34 @@
 import static java.util.stream.Stream.concat;
 
 /**
- * HTTP request with a fully aggregated/decoded body.
+ * An immutable HTTP request object containing a finite size content body in full.
+ * <p>
+ * A {@code FullHttpRequest} is useful for HTTP request messages with
+ * finite body content such as PUT or POST for creating or modifying a RESTful
+ * object.
+ * <p>
+ * FullHttpRequest is immutable. Once created, it is impossible to modify any
+ * request attributes, including body content. However the request can
+ * be transformed to another via {@code FullHttpRequest.Builder}.
```

Comment:
This seems a little disconnected from the text that explains the `newBuilder` method.

Reply:
@kvosper @dvlato @VivianLopes WDYT?

```
/**
 * An immutable HTTP request object including full body content.
 * <p>
 * A {@link FullHttpRequest} is useful for request messages with a
 * finite body content, such as when PUT or POST are used to create or
 * modify a RESTful object.
 * <p>
 * A FullHttpRequest is created via {@link FullHttpRequest.Builder}. A new builder
 * can be obtained by a call to following static methods:
 *
 * <ul>
 *     <li>{@code get}</li>
 *     <li>{@code head}</li>
 *     <li>{@code post}</li>
 *     <li>{@code put}</li>
 *     <li>{@code delete}</li>
 *     <li>{@code patch}</li>
 * </ul>
 *
 * A builder can also be created with one of the {@code Builder} constructors.
 *
 * FullHttpRequest is immutable. Once created it cannot be modified.
 * However a request can be transformed to another using the {@link this#newBuilder}
 * method. It creates a new {@link Builder} with all message properties and 
 * body content cloned in.
 */
```